src/core/DataSourceBase.ts

```diff
@@ -52,16 +62,12 @@ export abstract class DataSourceBase<TQuery extends DataQuery, TOptions extends
    *
    * @template T - The expected response type.
    * @param url - The endpoint URL for the GET request.
-   * @param params - Optional query parameters as a key-value map.
-   * @param useApiIngress - If true, uses API ingress bypassing the UI ingress for the request.
+   * @param options - Optional configurations for the request.
    * @returns A promise resolving to the response of type `T`.
    */
-  public async get<T>(url: string, params?: Record<string, any>, useApiIngress = false) {
-    if (useApiIngress) {
-      [url, params] = await this.buildApiRequestConfig(url, params ?? {}, 'GET');
-    }
-
-    return get<T>(this.backendSrv, url, params);
+  public async get<T>(url: string, options: RequestOptions = {}) {
```

Comment:
fyi this pr introduces new methods that should follow this pattern: https://github.com/ni/systemlink-grafana-plugins/pull/430/files#diff-c61a82d3fcf1fc318c4d2e8511f7a7b4088b905dbc3a8d8ae787d2dd4fd507b7R84